client/my-sites/upgrades/checkout/credit-card-payment-box.jsx

```diff
@@ -12,17 +13,69 @@ var PayButton = require( './pay-button' ),
 	TermsOfService = require( './terms-of-service' ),
 	PaymentBox = require( './payment-box' ),
 	analytics = require( 'lib/analytics' ),
-	cartValues = require( 'lib/cart-values' );
+	cartValues = require( 'lib/cart-values' ),
+	transactionStepTypes = require( 'lib/store-transactions/step-types' );
 
 import CartCoupon from 'my-sites/upgrades/cart/cart-coupon';
 import PaymentChatButton from './payment-chat-button';
 import config from 'config';
 import { PLAN_BUSINESS } from 'lib/plans/constants';
-import { some } from 'lodash';
+import ProgressBar from 'components/progress-bar';
 
 var CreditCardPaymentBox = React.createClass( {
 	getInitialState: function() {
-		return { previousCart: null };
+		return {
+			elapsed: 0,
+			previousCart: null
+		};
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( nextProps.transactionStep && this.submitting( nextProps.transactionStep ) ) {
```

Comment:
It probably doesn't affect the behavior of `this.timer`, but I believe that we will create the timer unnecessarily here, as this will fire multiple times for values of `nextProps.transactionStep` that return true from `this.submitting`. We can prevent this by comparing the existing `transactionStep` to the next one.